spec/dap/proto/ldap_proto_spec.rb

```diff
@@ -237,8 +237,8 @@
     end
 
     context 'testing empty ASN.1 Sequence' do
-      hex = ['308400000000']
-      data = OpenSSL::ASN1.decode(hex.pack('H*'))
+
+      data = OpenSSL::ASN1::Sequence.new([])
```

Comment:
Under 2.3.1 they both generate the same structure:

````
#<OpenSSL::ASN1::Sequence:0x007ff3c202ea00
 @infinite_length=false,
 @tag=16,
 @tag_class=:UNIVERSAL,
 @tagging=nil,
 @value=[]>
````

The hex was reused from a test against `.split_message` as it might also see the structure on the first pass at the data.  In the case of `.split_message` it doesn't use `OpenSSL::ASN.decode` so the problem with OpenSSL parsing these bytes will not occur and the test is still representative of actual code execution against live data.    

For `.parse_message` (tested above) OpenSSL::ASN1.decode is called against the data prior to it being passed.  This call is wrapped begin/rescue to handle any errors so `.parse_message` should only see properly decoded data and so the test above is representative of execution against live data.